app/services/show_challenge.rb

```diff
@@ -0,0 +1,41 @@
+require_relative '../repositories/repository_challenge'
+require 'forwardable'
+
+# retieve specific data for show page
+# avoid loading all 'entries' to limit memory
+class ShowChallenge
```

Comment:
Can't say I'm in love with the name, but struggling to come up with better alternatives.. Any others that you were considering?